app/views/users/show.html.erb

```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```

Comment:
You might want to create a different policy here, since `show?` handles who can render this page, this conditional never gets called.